### ISSUE TITLE
d3.svg.brush and mouseover event conflict

### ISSUE THREAD

**Issue by majorye** [d3/d3#1604]
Hi, I apply d3.svg.brush on some element, but I want to use mouseover event 
for the element in the brush area , so I could not get the mouseover event callback, 
how to solve this problem ?

**Comment by jwildfire** [comment]
[This example](https://bl.ocks.org/larsenmtl/4c5ebcc9bca489f8f530bd163b35f39b) from @larsenmtl takes a slightly different approach that worked for my use case.

**Comment by barry-johnson** [comment]
For those visiting in the future. I found the strategy illustrated by [this Plunkr](http://plnkr.co/edit/JDP4xJMbCEggxDdWiaK2) to work reasonably well. Original was from an answer to a [StackOverflow question](https://stackoverflow.com/questions/38670322/d3-brushing-and-mouse-move-coexist) although I corrected a small bug that prevented the mouseover horizontal bar from being drawn.

The key region is to view is lines 258-280 . After creating the brush, grab its background and attach additional mouseover/move/out events to the background.

The one notable limitation as implemented is that as-is it doesn't do the mouseover magic on the area inside an active brush, however I believe. additional mouseover events could be added to the active brush region to accomplish this.
